seahub/api2/endpoints/revision_tag.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2012-2016 Seafile Ltd.
+import re
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import status
+
+from django.utils.translation import ugettext as _
+
+from seahub.api2.throttling import UserRateThrottle
+from seahub.api2.authentication import TokenAuthentication
+from seahub.api2.utils import api_error
+from seahub.revision_tag.models import Tags, RevisionTags
+from seahub.views import check_folder_permission
+
+import seaserv
+from seaserv import seafile_api
+
+
+def check_parameter(func):
+    def _decorated(view, request, *args, **kwargs):
+        if request.method == "POST":
+            repo_id = request.data.get('repo_id', '')
+            tag_names = request.data.get('tag_names', '')
+            if not tag_names:
+                error_msg = "Tag can not be empty"
+                return api_error(status.HTTP_400_BAD_REQUEST, error_msg)
+            names = [name.strip() for name in tag_names.split(',')]
+            tag_names = []
+            for name in names:
+                if not check_tagname(name):
+                    error_msg = u'Tag can only contain letters, numbers, dot, hyphen or underscore'
```

Comment:
Tag can only `contains` letters, numbers, dot, hyphen or underscore. Translation is needed.